src/board.cpp

```diff
@@ -0,0 +1,166 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include "board.h"
+#include "gui.h"
+
+Board::Board() : Board(9, false) {}
+
+Board::Board(size_t size, bool cpu) {
+  isCpu = cpu;
+  p1Score = 0;
+  p2Score = 0;
+  boardSize = size;
+  for (int i = 1; i <= size; i++) {
+    p1.insert(i);
+    p2.insert(i);
+  }
+}
+
+int Board::move(int player, int number) {
+  std::set<int>& playerSet = player == 1 ? p1 : p2;
+  if (playerSet.find(number) == playerSet.end()) {
+    return -1;
+  }
+  playerSet.erase(number);
+  return number;
+}
+
+void Board::playMove(int p1Move, int p2Move) {
+  move(P1, p1Move);
+  move(P2, p2Move);
+  if (p1Move > p2Move) {
+    p1Score++;
+  } else if (p2Move > p1Move) {
+    p2Score++;
+  }
+}
+
+bool Board::isValidMove(int player, int number) {
+  const std::set<int>& playerSet = player == P1 ? p1 : p2;
+  return playerSet.find(number) != playerSet.end();
+}
+
+std::string Board::getPlayerMoves(int player) {
+  const std::set<int>& playerSet = player == P1 ? p1 : p2;
```

Comment:
```suggestion
  const std::set<int>& playerSet = ((player == P1) ? p1 : p2);
```
This should be a macro

Reply:
yeah perhaps I should just abstract this out to a class :)